cluster-logging.md

```diff
@@ -0,0 +1,110 @@
+# Cluster Logging and OpenTelemetry
+
+This is the protocol and semantic conventions documentation for Red Hat OpenShift Logging.
+
+## Forwarding and Ingestion Protocol
+
+Red Hat OpenShift Logging provides a log collection and forwarding solution that is capable of writing logs to OpenTelemetry endpoints using OTLP. [OTLP](https://opentelemetry.io/docs/specs/otlp/) is the *protocol* for encoding, transporting, and delivering telemetry data.  This product additionally is capable of providing a Loki storage deployment that provides an OTLP endpont to ingest log streams.  This document defines the semantic conventions associated with the logs collected from the various sources of an OpenShift cluster.
+
+**Note:** Logs are forwarded using OTLP/HTTP as defined by the OpenTelemetry Observability Framework.  It uses Protobuf payloads encoded in JSON format.
+
+
+## Semantic Conventions
+
+The log collector provided by this solution collects the following log streams:
+
+* Container logs
+* Cluster node journal logs
+* Cluster node auditd logs 
+* Kubernetes and OpenShift API server logs
+* OpenShift Virtual Network (OVN) logs
+
+These streams are forwarded using the sementatic conventions defined by [OpenTelemetry semantic attributes](https://github.com/open-telemetry/semantic-conventions/tree/main/docs). The semantic conventions in OpenTelemetry define a *Resource* as an immutable representation of the entity producing telemetry as *Attributes*. For example, a process producing telemetry that is running in a container has a container_name, a cluster_id, a pod_name, a namespace, and possibly a deployment or app_name. All of these *Attributes* are included in the *Resource* object.  This grouping and reducing of common attributes is a powerful tool when sending logs as telemetry data.
+
+
+The following sections define the attributes that are forwarded.
+
+
+### Common Attributes
+
+All log streams include a minimal set of attributes.
+
+Resource Attributes:
+```
+cluster.id
+node.name
+openshift.log.source
+openshift.log.type
+```
+
+Log Attributes:
+```
+body
+timeUnixNano
```

Comment:
@alanconway what is the "right" value to provide here?  [This](https://opentelemetry.io/docs/specs/otel/logs/data-model/#log-and-event-record-definition) seems to say timestamp which is different from the protobuf [spec](https://pkg.go.dev/go.opentelemetry.io/proto/otlp/logs/v1#pkg-types).  did it change?  Is it OTEL spec version specific?  Do we need to identify the OTEL version to which we adhere ?